### DISCUSSION TITLE
Transfer payment from Asset Account to Credit Card Asset Account

### DISCUSSION THREAD

**Discussion by tatleoat** [firefly-iii/firefly-iii#9325] [Firefly III]
Hello,

So I'm having an issue with the transfer functionality in firefly iii, here's what I understand:

Account 1: asset account
Account 2: credit card account

1. Since credit card accounts are expense accounts (the csv I downloaded from discover is structure like this) the expenses accrue positively while the payments are negative in account 2, which is opposite from account 1
2. When I try to use the transfer function from my asset account, it subtracts money from account 1 and reflexively tries to add That sum to account 2.
3. The payments need to be negative, yet the nature of transfer forces a positive number into the credit card.

In short, the transfer is subtracting from my asset account, and you'd expect it to also subtract from the credit account since it's an expense account but it just adds to it as if the payment was a purchase.

What am I doing wrong? I've looked through all the documentation and posts I could find but nothing explains this discrepancy. Thank you in advance.

**Comment by tatleoat** [comment]
![image](https://github.com/user-attachments/assets/5e97dcfd-4319-4918-8f77-37a66993da42)

A visual representation would probably be helpful. This is my Credit Card Account, set up as an Asset Account. 

The first three transactions are transfers from my Fifth Third acct and are intended to be transfers to my Discover Card acct. The 5th transfer is a Withdrawal, sent to the Discover Expense Acct as a payment. All the remaining transactions are, while still technically positive numbers, purchases. Adding more purchases increases the total balance, which makes sense. 

I expected a transfer from the Fifth Third acct to the Discover would decrease, but it also increases it as well. So both making a purchase and transferring a payment increase the total of the Discover Card.

I apologize if I'm being confusing or not as helpful as I could be, I'm new to the world of finance (and linux) but I'm trying really hard and I really appreciate your software

**Reply by JC5** [reply]
It looks like the last two transactions have been stored as deposits, not as withdrawals. It may depend on your CSV file if they can be recognized properly. Could you share some example lines and your import configuration?

**Comment by tatleoat** [comment]
I would be happy to provide that information for you :)

Here is a small snippet of the CSV, downloaded directly from the Discover website

![image](https://github.com/user-attachments/assets/0c3ea284-d413-4dca-b8d8-f1cb4e547d66)

And here is the config file

[import_config_2024-10-08.json](https://github.com/user-attachments/files/17296287/import_config_2024-10-08.json)

I want to take a second to address what you're saying about the transactions being regarded as deposits, not withdrawals. I think I understand where I'm getting confused now, so you have a message here:

https://github.com/orgs/firefly-iii/discussions/8589#discussioncomment-8596269

The OP stated in their question "My expense account that belongs to my credit card shows a positive balance, and I don't think that should be right? Maybe?" and you responded with "Yes, expense accounts get a positive balance, that's normal. So no worries there." 

I understood that to mean that charges to my credit card were to be regarded as positive numbers contributing positively to the Credit total, and this was reaffirmed by the fact those charges in my CSV were natively recorded as positive numbers, is this not typically the case?

**Reply by JC5** [reply]
Do you have the raw data, and not a screenshot? You can also send me the data: james@firefly-iii.org

**Reply by tatleoat** [reply]
> Do you have the raw data, and not a screenshot? You can also send me the data: [james@firefly-iii.org](mailto:james@firefly-iii.org)

Sorry, I misunderstood, sent :)

**Reply by JC5** [reply]
I have them, thanks for your patience.

I'm not sure if I can fix this for you. I see too little information in your CSV files to determine if it's a withdrawal or a transfer or what.
